Update "Deep Dive" in reusing-logic-with-custom-hooks.md with link and example of the use API (#7767)
reactjs/react.dev: src/content/learn/reusing-logic-with-custom-hooks.md

#### Will React provide any built-in solution for data fetching? {/*will-react-provide-any-built-in-solution-for-data-fetching*/}

> Today, with the [`use`](/reference/react/use#streaming-data-from-server-to-client) API, data can be streamed from the server to the client by passing a [Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) as a prop:

Reviewer: It doesn't necessarily need to be streamed from the server, it just needs to be an existing cached promise. 

Maybe something like this:

```suggestion
Today, with the [`use`](/reference/react/use#streaming-data-from-server-to-client) API, data can be read in render by passing a [Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) to `use`:
```
With a link to the `use` reference docs?

#### Will React provide any built-in solution for data fetching? {/*will-react-provide-any-built-in-solution-for-data-fetching*/}

> Today, with the [`use`](/reference/react/use#streaming-data-from-server-to-client) API, data can be streamed from the server to the client by passing a Promise as a prop from a [Server Component](/reference/rsc/server-components) and resolving it in a [Client Component](/reference/rsc/use-client):

Reviewer: This kinda makes it sound like you can only use `use` with RSC

Author: I agree. The [docs](https://react.dev/reference/react/use#streaming-data-from-server-to-client) for "usage" only specify the RSC example. Would it be better to reference the more generic example [here](https://react.dev/reference/react/use#reference)?
